index.js

```diff
@@ -393,6 +395,27 @@ async function fastifyStatic (fastify, opts) {
     pumpSendToReply(req, reply, file, rootPath)
   }
 }
+function convertRootPathIfUrl (rootPath) {
+  const convertIfUrl = (path) => {
```

Comment:
![image](https://user-images.githubusercontent.com/5059100/224479959-d5319331-6bd5-4cc5-9618-b332b86e6a73.png)

Reply:
The code coverage was failing for inline map lmao, couldn't get it full coverage without

Reply:
Am I supposed to make a change here?

Reply:
I am currently recovering from a flu. So my brain works at about 50% capacity.

I would prefer a function, which always returns an Array of strings. 
I would not use Array.map() functions as they suck.

Reply:
> I would prefer it like this. Not tested. Programmed on my phone...
> 
> ```js
> function normalizeRoot(root) {
>   const result = []
> 
>   if (typeof root === 'string') {
>     result.push(root)
>   } else if (root instanceof URL && root.protocol === 'file:') {
>     result.push(url.fileURLToPath(root))
>   } else if (Array.isArray(root)) {
>     for (let i = 0, il = root.length; i < il; ++i) {
>       if (typeof root === 'string') {
>         result.push(root)
>       } else if (root instanceof URL && root.protocol === 'file:') {
>         result.push(url.fileURLToPath(root)
>       } else {
>         throw new Error('Invalid root value')
>       }
>     }
>   } else {
>     throw new Error('Invalid root value')
>   }
> 
>   return result
> }
> ```

Why only check for file protocol?

Reply:
Are other protocols supported?

Reply:
Unsure, I was basing this off of: https://github.com/fastify/fastify-static/issues/364

Btw, on [index.js:413](https://github.com/fastify/fastify-static/blob/master/index.js#L413)  it uses map `rootPath.map((path) => checkPath(fastify, path))`

Should we change that as well then? Or should that be in a separate PR.

Reply:
I updated the code. There's no need to check if it's a string and such in the normalization because we have a function that does that already. `checkRootPathForErrors`. IMO these two functions do two different things and it's good to have separation of concerns.